services/director-v2/tests/unit/test_computation_utils.py

```diff
@@ -0,0 +1,200 @@
+        ),
+        (
+            # pipeline is published if any of the node is published AND time is within publication timeout
+            [
+                (
+                    RunningState.PUBLISHED,
+                    "datetime.utcnow()-timedelta(seconds=CELERY_PUBLICATION_TIMEOUT + 75)",
+                ),
+                (RunningState.PUBLISHED, "datetime.utcnow()-timedelta(seconds=145)"),
+                (RunningState.PUBLISHED, "datetime.utcnow()-timedelta(seconds=1555)"),
+            ],
+            RunningState.NOT_STARTED,
+        ),
+        (
+            # not started pipeline (all nodes are in non started mode)
+            [
+                (
+                    RunningState.NOT_STARTED,
+                    "fake.date_time()",
+                ),
+                (
+                    RunningState.NOT_STARTED,
+                    "fake.date_time()",
+                ),
+            ],
+            RunningState.NOT_STARTED,
+        ),
+        (
+            # successful pipeline if ALL of the node are successful
+            [
+                (RunningState.SUCCESS, "fake.date_time()"),
+                (RunningState.SUCCESS, "fake.date_time()"),
+            ],
+            RunningState.SUCCESS,
+        ),
+        (
+            # pending pipeline if ALL of the node are pending
+            [
+                (RunningState.PENDING, "fake.date_time()"),
+                (RunningState.PENDING, "fake.date_time()"),
+            ],
+            RunningState.PENDING,
+        ),
+        (
+            # failed pipeline if any of the node is failed
+            [
+                (RunningState.PENDING, "fake.date_time()"),
+                (RunningState.FAILED, "fake.date_time()"),
+                (RunningState.PENDING, "fake.date_time()"),
+            ],
+            RunningState.FAILED,
+        ),
+        (
+            # started pipeline if any of the node is started
+            [
+                (RunningState.STARTED, "fake.date_time()"),
+                (RunningState.FAILED, "fake.date_time()"),
+            ],
+            RunningState.FAILED,
+        ),
+        (
+            # started pipeline if any of the node is started
+            [
+                (RunningState.SUCCESS, "fake.date_time()"),
+                (RunningState.PENDING, "fake.date_time()"),
+                (RunningState.PENDING, "fake.date_time()"),
+            ],
+            RunningState.STARTED,
+        ),
+        (
+            # ABORTED pipeline if any of the node is aborted
+            [
+                (RunningState.SUCCESS, "fake.date_time()"),
+                (RunningState.ABORTED, "fake.date_time()"),
+                (RunningState.PENDING, "fake.date_time()"),
+            ],
+            RunningState.ABORTED,
+        ),
+        (
+            # empty tasks (could be an empty project or filled with dynamic services)
+            [],
+            RunningState.NOT_STARTED,
+        ),
+    ],
```

Comment:
Would you mind creating some named tuples or something to make it more easy to follow when reading?